spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteManifestsProcedure.java

```diff
@@ -331,4 +331,33 @@ public void testReplacePartitionField() {
             row(1, Timestamp.valueOf("2022-01-01 10:00:00"), Date.valueOf("2022-01-01"))),
         sql("SELECT * FROM %s WHERE ts < current_timestamp()", tableName));
   }
+
+  @Test
+  public void testWriteManifestWithSpecId() {
+    sql(
+        "CREATE TABLE %s (id int, dt string, hr string) USING iceberg PARTITIONED BY (dt)",
+        tableName);
+
+    sql("INSERT INTO %s VALUES (1, '2024-01-01', '00')", tableName);
+    sql("INSERT INTO %s VALUES (2, '2024-01-01', '00')", tableName);
+    assertEquals(
+        "Should have 2 manifests",
+        ImmutableList.of(row(2L)),
+        sql("SELECT count(*) FROM %s.manifests", tableName));
+
+    sql("ALTER TABLE %s ADD PARTITION FIELD hr", tableName);
```

Comment:
Could you please add more rows here and prove that you then have manifests from multiple spec's so that you can prove the specification only combined manifests from the correctly specified spec

Reply:
Addressed comment.